api/web3server.go

```diff
@@ -909,30 +914,107 @@ func (svr *web3Handler) traceTransaction(ctx context.Context, in *gjson.Result)
 		return nil, err
 	}
 
-	structLogs := make([]structLog, 0)
-	for _, s := range traces.StructLogs() {
-		var enc structLog
-		enc.Pc = s.Pc
-		enc.Op = s.Op
-		enc.Gas = math.HexOrDecimal64(s.Gas)
-		enc.GasCost = math.HexOrDecimal64(s.GasCost)
-		enc.Memory = s.Memory
-		enc.MemorySize = s.MemorySize
-		enc.Stack = s.Stack
-		enc.ReturnData = s.ReturnData
-		enc.Storage = s.Storage
-		enc.Depth = s.Depth
-		enc.RefundCounter = s.RefundCounter
-		enc.OpName = s.OpName()
-		enc.ErrorString = s.ErrorString()
-		structLogs = append(structLogs, enc)
+	return &debugTraceTransactionResult{
+		Failed:      receipt.Status != uint64(iotextypes.ReceiptStatus_Success),
+		Revert:      receipt.ExecutionRevertMsg(),
+		ReturnValue: byteToHex(retval),
+		StructLogs:  fromLoggerStructLogs(traces.StructLogs()),
+		Gas:         receipt.GasConsumed,
+	}, nil
+}
+
+func (svr *web3Handler) traceCall(ctx context.Context, in *gjson.Result) (interface{}, error) {
+	var (
+		blkHash      hash.Hash256
+		err          error
+		contractAddr string
+		callData     []byte
+		gasLimit     uint64
+		value        *big.Int
+		callerAddr   address.Address
+		gasPrice     int64
+	)
+	core := svr.coreService.(*coreService)
```

Comment:
use funcs in the `CoreService` interface

Reply:
traceCall needs to access `core.bc`, `core.dao`,`core.sf`,use interface can not be